### ISSUE TITLE
prefer client language for warning if set

### ISSUE THREAD

**Issue by DanielRuf** [phpMussel/phpMussel#28]
I searched for `$_SERVER['HTTP_ACCEPT_LANGUAGE']` in the code but did not find it.

We could add a setting to prefer the language of the client (language set in the browser) and use this for the warning shown to visitors/users to make it more flexible.

**Comment by Maikuolan** [comment]
Agreed. Working on this now.

**Comment by Maikuolan** [comment]
Committed. :-)

**Comment by DanielRuf** [comment]
Great, I will test this tomorrow.

**Comment by DanielRuf** [comment]
Changing the language works so far.

But there is an issue with the scanlog. If I scan with the german language, the scanlog saves a line with the german `detected` string, but this should be language independent.

Because when I change now my language to another one and I rescan the same file, phpMussel uses the german string from the scanlog

https://github.com/Maikuolan/phpMussel/blob/master/vault/phpmussel.inc#L12404

```
if($GLOBALS['MusselConfig']['general']['scan_log']&&$n!=0)$s=date('r').' '.$GLOBALS['MusselConfig']['lang']['started'].'.'.$GLOBALS['linebreak'];
$r=phpMusselR($f,$n,$zz,$dpt,$ofn);
```

$s is the new string without the `detected`and filename parts and $r is the old one from the scanlog file but with german text. So I get a german `detected` text when I am using another language.

**Comment by Maikuolan** [comment]
Hmm, I hadn't considered that when making the change. :-/

Good spotting. I may need to rethink how I save log data; I'll have a think about it and try to come up with something.

**Comment by DanielRuf** [comment]
@Maikuolan We could use some array and serialize / unserialize it (store it serialized, so we have just one line in the logfile and can extract it for showing in the frontend without any problem. Some elements/values could contain some id representing the result (detected x ...).

Or just one line as a string, not serialized array.
Should be better to read and analyze (think about delimiting the values with comma or tab or something else, so it can be easily inserted into spreadsheet software like Excel and easily analyzed).

But this could be another ticket/issue (custom log format).

**Comment by Maikuolan** [comment]
New issue created. :-)

https://github.com/Maikuolan/phpMussel/issues/54

**Comment by Maikuolan** [comment]
*A quick note regarding this issue: Issue is not dead, but has been delayed for now, until other higher priority issues can be resolved. Changes originally implemented since first creating this issue (i.e., the feature in question) had been redacted due to the aforementioned bug whereby we'd end up with the wrong L10N being used for logs, wrong languages being displayed to the client due to caching and so on. The package is currently free of this bug, due to that it's also currently free of the feature that the bug is associated with. Reinstating this feature is still on the to-do list, but requires changes, as to not reinstate this bug along with it.*

__Current progress:__

In `/vault/lang.php`, you'll notice these lines of PHP code:
```PHP
/** Will remove later (temporary variable). */
$phpMussel['Config']['general']['lang_override'] = false;
```

That "temporary variable" serves to define a currently non-existent configuration directive, intended to enable/disable preferring L10N data as per the language specified by the client (when available), above the default language as per defined by the configuration, and will be removed in the future, when this configuration option is reinstated.

Directly below those lines is some code for fetching the necessary L10N, both client-defined and configuration-defined, available respectively via `$phpMussel['lang_user']` and `$phpMussel['lang']`.

I was thinking of taking the approach whereby we can use the configuration-defined L10N data for writing to the logs, configuration, etc (basically, anything that the client won't directly see for themselves), and then the client-defined L10N data for the things that the client will see. That all still needs to be coded though (other than the code in `/vault/lang.php`, not much has been done yet in the package towards properly resolving this issue).

__To do:__

- [x] Write code to load client-defined and configuration-defined L10N data and make both available to the instance.
- [x] Add temporary switch for enabling/disabling feature.
- [x] Form a distinction between L10N data served to the client versus other L10N data.
- [x] Cached scan results and other cache data should ideally be language-agnostic wherever possible. Need to make changes to the code accordingly. Should revert to configuration-defined L10N data wherever not possible (backwards-compatibility issues, etc).
- [x] Modify codebase as to make use of the distinction between client-defined and configuration-defined L10N data.
- [x] Reinstate `lang_override` configuration directive, remove temporary switch, and update documentation.
- [x] Test changes to confirm working as expected, and when confirmed, close issue.

**Comment by Maikuolan** [comment]
TL;DR -- Everything done now. Marking as fixed/implemented and closing (46e86e0e42eb233083e544da7f2e3168e033da8e and ab5aba5e538f0f6d73a36eec1b123943b65a82f7).
